blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/UserImpl.java

```diff
@@ -143,6 +151,43 @@ public Link getLink() {
         return (parent != null)?parent.getLink().rel(getId()): ApiHead.INSTANCE().getLink().rel("users/"+getId());
     }
 
+    /**
+     * Gets or creates the user's private Jenkins-managed key and returns the
+     * public key to the user
+     * @return JSON response
+     * @throws IOException 
+     */
+    @GET
+    @WebMethod(name="publickey")
+    public HttpResponse publicKey() throws IOException {
+        User authenticatedUser =  User.current();
+        if(authenticatedUser == null || !StringUtils.equals(getId(), authenticatedUser.getId())) {
+            throw new ServiceException.UnauthorizedException("Not authorized");
+        }
+        
+        String publicKey = UserSSHKeyManager.getReadablePublicKey(authenticatedUser, 
+            UserSSHKeyManager.getOrCreate(authenticatedUser));
+        
+        return HttpResponses.okJSON(ImmutableMap.of("key", publicKey));
+    }
+
+    /**
+     * Deletes the user's private Jenkins-managed key
+     * @return
+     * @throws IOException 
+     */
+    @DELETE
+    @WebMethod(name="publickey")
+    public HttpResponse resetPublicKey() throws IOException {
+        User authenticatedUser =  User.current();
+        if(authenticatedUser == null || !StringUtils.equals(getId(), authenticatedUser.getId())) {
+            throw new ServiceException.UnauthorizedException("Not authorized");
```

Comment:
Hmm, I think it should be split to throw UnauthorizedException if authenticatedUser is null, if user ids do not match then Forbidden. Basically, Unauthorized or 401, which tells client to ask user to login/authenticate as authentication info was not found. Forbidden is 403, which is to tell client its unauthorized access and user is not allowed to access the resource.